Extension/src/LanguageServer/configurations.ts

```diff
@@ -652,6 +652,16 @@ export class CppProperties {
                         configuration.macFrameworkPath = this.defaultFrameworks;
                     }
                 }
+            } else {
+                // However, if compileCommands are used and compilerPath is explicitly set, it's still necessary to resolve variables in it.
+                if (configuration.compilerPath === null) {
+                    configuration.compilerPath = undefined;
+                } else {
+                    if (configuration.compilerPath === "${default}") {
+                        configuration.compilerPath = settings.defaultCompilerPath;
```

Comment:
I think we may need to handle the case where settings.defaultCompilerPath is null/undefined?

Reply:
Should this check be moved to be the 1st checked?